docker-compose.yml

```diff
@@ -5,24 +5,105 @@ version: '3'
 services:
   zookeeper:
     image: confluentinc/cp-zookeeper:5.0.1
-    ports:
-      - 2181:2181
     restart: unless-stopped
     environment:
-      ZOOKEEPER_CLIENT_PORT: 2181
-      ZOOKEEPER_TICK_TIME: 2000
+      ZOOKEEPER_SERVER_ID: 1
+      ZOOKEEPER_CLIENT_PORT: "2181"
+      ZOOKEEPER_TICK_TIME: "2000"
+    network_mode: "host"
 
-  kafka:
+  kafka1:
     image: confluentinc/cp-kafka:5.0.1
-    ports:
-      - 9092:9092
-    restart: unless-stopped
     depends_on:
       - zookeeper
     environment:
+      KAFKA_ZOOKEEPER_CONNECT: "localhost:2181"
+      KAFKA_LISTENER_SECURITY_PROTOCOL_MAP: PLAINTEXT:PLAINTEXT,PLAINTEXT_HOST:PLAINTEXT
+      KAFKA_INTER_BROKER_LISTENER_NAME: PLAINTEXT
+      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://localhost:9091,PLAINTEXT_HOST://localhost:29091
+      KAFKA_LOG4J_LOGGERS: "kafka.authorizer.logger=INFO"
       KAFKA_BROKER_ID: 1
-      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
-      KAFKA_ADVERTISED_LISTENERS: PLAINTEXT://localhost:9092
-      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 1
-      KAFKA_NUM_PARTITIONS: 8
+      KAFKA_BROKER_RACK: "r1"
+      KAFKA_OFFSETS_TOPIC_REPLICATION_FACTOR: 3
+      KAFKA_DELETE_TOPIC_ENABLE: "true"
```

Comment:
Perhaps the common envs could be moved to a separate file and included with `env_file`.

Reply:
21b567f